src/renderer/dx/DxFontRenderData.cpp

```diff
@@ -441,6 +446,77 @@ try
 }
 CATCH_RETURN()
 
+// Routine Description:
+// - Returns whether the user set or updated any of the font features to be applied
+bool DxFontRenderData::DidUserSetFeatures() const noexcept
+{
+    return _didUserSetFeatures;
+}
+
+// Routine Description:
+// - Clears our internal feature map and populates it with the list of standard features
+void DxFontRenderData::InitializeFeatureMap()
+{
+    _featureMap.clear();
+    _featureMap = {
+        { L"rlig", 1 },
+        { L"rclt", 1 },
+        { L"locl", 1 },
+        { L"ccmp", 1 },
+        { L"calt", 1 },
+        { L"liga", 1 },
+        { L"clig", 1 },
+        { L"kern", 1 },
+        { L"mark", 1 },
+        { L"mkmk", 1 },
+        { L"dist", 1 }
+    };
+}
+
+// Routine Description:
+// - Updates our internal map of font features with the given features
+// Arguments:
+// - features - the features to update our map with
+void DxFontRenderData::SetFeatures(std::unordered_map<std::wstring_view, uint32_t> features)
+{
+    // Populate our feature map with the standard list
+    InitializeFeatureMap();
+
+    // Update our feature map with the provided features
+    if (!features.empty())
+    {
+        for (const auto [tag, param] : features)
+        {
+            _featureMap[tag] = param;

```

Comment:
Actually, should we be tossing out any of these who's `tag` _isn't_ 4 characters? Users may typo a font feature, and then we probably shouldn't crash

Reply:
Yep, thanks!